streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/TaskAssignorConvergenceTest.java

```diff
@@ -45,36 +46,29 @@
 
         private static Harness initializeCluster(final int numStatelessTasks,
                                                  final int numStatefulTasks,
-                                                 final int numNodes) {
+                                                 final int numNodes,
+                                                 final Supplier<Integer> partitionCountSupplier) {
             int subtopology = 0;
             final Set<TaskId> statelessTasks = new TreeSet<>();
-            {
-                int partition = 0;
-                for (int i = 0; i < numStatelessTasks; i++) {
-                    statelessTasks.add(new TaskId(subtopology, partition));
-                    if (partition == 4) {
-                        subtopology++;
-                        partition = 0;
-                    } else {
-                        partition++;
-                    }
+            int remainingStatelessTasks = numStatelessTasks;
+            while (remainingStatelessTasks > 0) {
+                final int partitions = Math.min(remainingStatelessTasks, partitionCountSupplier.get());
```

Comment:
I was able to reproduce some more edge cases by not creating exactly 4 partitions per subtopology. Now, we randomly generate the number of partitions for each subtopology. There's a supplier instead of a PRNG because for the specific scenarios, we still want a fixed partition scheme.